cpp/src/arrow/csv/reader.cc

```diff
@@ -1166,10 +1313,26 @@ Future<std::shared_ptr<StreamingReader>> MakeStreamingReader(
   });
 }
 
+Future<std::shared_ptr<StreamingReader>> MakeStreamingReader(
```

Comment:
I'll defer to the input of others but I think an overload of `MakeStreamingReader` based solely on the type of `input` is a rather subtle API.  Could we maybe just rename this to `MakeStreamingFileReader`?

Reply:
I think it's ok :-) these are all internal APIs.